### PR TITLE
Complete Mad Libs assignment

### PR DESCRIPTION
**Pull request by RyanSpittler**
[#88836354]

### CODE REVIEW

File: methods_and_recursion/mad_libs.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env ruby
+
+# This helps to check whether 'a' or 'an' will be used.
+class String
+  def vowel?
```
Comment (jasonnoble) [inline methods_and_recursion/mad_libs.rb]: Please don't monkey patch for these assignments...

File: methods_and_recursion/mad_libs.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    vowels = %w(a e i o u A E I O U)
+    if vowels.include?(self)
```
Comment (jasonnoble) [inline methods_and_recursion/mad_libs.rb]: Pro Tip:

```
if ________
  true
else
  false
end
```

can be rewritten as:

```
___________
```

File: methods_and_recursion/mad_libs.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      true
+    else
+      false
+    end
+  end
+end
+
+def ask_for_word(term, list_of_words)
```
Comment (jasonnoble) [inline methods_and_recursion/mad_libs.rb]: How about an optional third parameter `seen_before` which defaults to false?

```
another_word = seen_before ? 'a' : 'another'
```

File: methods_and_recursion/mad_libs.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                 'part of the body',
+                 'place',
+                 "verb ending in 'ing'",
+                 'adjective',
+                 'number',
+                 'plural noun'
+                ]
+
+needed_blanks.each do |term|
+  mad_libs << ask_for_word(term, already_requested)
```
Comment (jasonnoble) [inline methods_and_recursion/mad_libs.rb:41]: Won't `already_requested` always be the empty array from line 27?  You need to add term to `already_requested` somewhere.

File: methods_and_recursion/mad_libs.rb [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+end
+
+puts "Here is your madlib:"
```
Comment (jasonnoble) [inline methods_and_recursion/mad_libs.rb]: You can DRY up your code a bit:

```
puts <<END_OF_MADLIB
  A vacation is when you take a trip to some #{mad_libs[0] place
  with your #{mad_libs[1]} family. Usually you go to some place
  ...
END_OF_MADLIB
```

**Comment by jasonnoble** [comment]
Rebase me...

File: methods_and_recursion/mad_libs.rb (hunk 4 above)
Comment (RyanSpittler) [inline methods_and_recursion/mad_libs.rb:41]: It was hidden on line 21 of the `ask_for_words` method.

**Pull request merged**